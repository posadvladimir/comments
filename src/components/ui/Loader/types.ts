
import type { LoaderSizes } from 'src/ts/enums/loader'

export type TLoaderProps = {
    size?: LoaderSizes;
}