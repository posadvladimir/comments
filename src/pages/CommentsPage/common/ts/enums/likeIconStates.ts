
export enum LikeIconStates {
    GreyStroke = 'grey-stroke',
    RedStroke = 'red-stroke',
    FullRed = 'full-red',
}