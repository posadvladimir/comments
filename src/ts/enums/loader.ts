
export enum LoaderSizes {
    Small = 20,
    Big = 35,
}