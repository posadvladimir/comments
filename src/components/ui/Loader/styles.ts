
import { LoaderSizes } from 'src/ts/enums/loader';

import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
`;

const SPINNER_WIDTH_DIVIDER = 5.5;

export const LoaderContainer = styled.div`
    display: flex;
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    justify-content: center;
    align-items: center;
`;

export const LoaderSpinner = styled.div<{ $size: LoaderSizes }>`
    content: " ";
    display: block;
    width: ${({ $size }) => $size}px;
    height: ${({ $size }) => $size}px;
    border-radius: 50%;
    border: ${({ $size }) => ($size / SPINNER_WIDTH_DIVIDER)}px solid #fff;
    border-color: #fff transparent #fff transparent;
    animation: ${rotate} 1.2s linear infinite;
`;