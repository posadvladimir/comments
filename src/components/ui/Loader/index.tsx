
import type { TLoaderProps } from './types';

import * as S from './styles';

import { LoaderSizes } from 'src/ts/enums/loader';

import { LOADER_TEST_ID } from 'src/constants/test-ids';

const Loader: React.FC<TLoaderProps> = ({ size = LoaderSizes.Small }) => (
    <S.LoaderContainer>
        <S.LoaderSpinner
            $size={size}
            data-testid={LOADER_TEST_ID}
        />
    </S.LoaderContainer>
);

export default Loader;