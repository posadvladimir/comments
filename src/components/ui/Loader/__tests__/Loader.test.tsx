
import { runSnapshotTest } from 'src/common/tests/snapshot';

import { LoaderSizes } from 'src/ts/enums/loader';

import Loader from '..';

describe('Loader Component', () => {
    runSnapshotTest(<Loader />);
    runSnapshotTest(<Loader size={LoaderSizes.Small} />);
    runSnapshotTest(<Loader size={LoaderSizes.Big} />);
});