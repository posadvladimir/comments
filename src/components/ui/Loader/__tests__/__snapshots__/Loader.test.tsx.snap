// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component соотвествует снепшоту 1`] = `
<div
  className="sc-bcPKhP uEXKN"
>
  <div
    className="sc-grXZZQ hkWoTP"
    data-testid="loader-test-id"
  />
</div>
`;

exports[`Loader Component соотвествует снепшоту 2`] = `
<div
  className="sc-bcPKhP uEXKN"
>
  <div
    className="sc-grXZZQ hkWoTP"
    data-testid="loader-test-id"
  />
</div>
`;

exports[`Loader Component соотвествует снепшоту 3`] = `
<div
  className="sc-bcPKhP uEXKN"
>
  <div
    className="sc-grXZZQ hfrGbl"
    data-testid="loader-test-id"
  />
</div>
`;
